expression/builtin_cast_vec.go

```diff
@@ -109,7 +109,7 @@ func (b *builtinCastIntAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.
 		}
 		if !hasUnsignedFlag0 && !hasUnsignedFlag1 {
 			rs[i] = float64(i64s[i])
-		} else if b.inUnion && i64s[i] < 0 {
+		} else if b.inUnion && !hasUnsignedFlag1 && i64s[i] < 0 {
```

Comment:
ditto

Reply:
It's in a fop loop, maybe it's acceptable here.

Reply:
@SunRunAway I am not mean the local variable, see: https://github.com/pingcap/tidb/pull/13637#discussion_r349977712, I mean this logic is not comprehensable.

Reply:
I do not get you. What is your suggestion?